File: LayoutTests/http/tests/misc/resources/image-checks-for-accept.php
<?php
    $allimages = 'image/*';
    $pos = strpos($_SERVER["HTTP_ACCEPT"], $allimages);
    if ($pos !== false)
    {
        header("Content-Type: image/jpg");
        header("Cache-Control: no-store");
        header("Connection: close");

        $fn = fopen("compass.jpg", "r");
        fpassthru($fn);
        fclose($fn);
        exit;
    }
?>

File: LayoutTests/http/tests/misc/resources/image-checks-for-accept.py
#!/usr/bin/env python3

import os
import sys

http_accept = os.environ.get('HTTP_ACCEPT', '')

if 'image/*' in http_accept:
    sys.stdout.write(
        'Content-Type: image/jpg\r\n'
        'Cache-Control: no-store\r\n'
        'Connection: close\r\n\r\n'
    )

    sys.stdout.flush()
    with open(os.path.join(os.path.dirname(__file__), 'compass.jpg'), 'rb') as fn:
        sys.stdout.buffer.write(fn.read())
else:
    sys.stdout.write('Content-Type: text/html\r\n\r\n')
